text/0000-my-feature.md

```diff
@@ -0,0 +1,113 @@
+# Meta
+[meta]: #meta
+- Name: Implementing pack detect command
+- Start Date: 2024-02-15
+- Author(s): @rashadism
+- Status: Draft <!-- Acceptable values: Draft, Approved, On Hold, Superseded -->
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: N/A
+
+# Summary
+[summary]: #summary
+
+The `pack detect` command is introduced to the Cloud Native Buildpacks ecosystem, providing a way to run only the detect phase of the buildpack lifecycle. This feature enhances the developer experience by allowing them to quickly determine which buildpacks are relevant for their application without progressing through the entire build process. This was partially discussed in [issue #681](https://github.com/buildpacks/pack/issues/681) but the issue was about implementing a `dry-run` flag. With further discussion with @jjbustamante, decided to go forward with this as a new pack pack command rather than a flag.
+
+# Definitions
+[definitions]: #definitions
+
+Make a list of the definitions that may be useful for those reviewing. Include phrases and words that buildpack authors or other interested parties may not be familiar with.
+
+# Motivation
+[motivation]: #motivation
+
+- Simplify and streamline the build process by providing a targeted command for buildpack detection.
+- Reduce build times by skipping unnecessary phases of the buildpack lifecycle.
+- Enable developers to quickly identify which buildpacks are applicable to their application without waiting for the entire build process to complete, or having to `Ctrl+C` midway through.
+
+# What it is
+[what-it-is]: #what-it-is
+
+This provides a high level overview of the feature.
+
+- Define any new terminology.
+- Define the target persona: buildpack author, buildpack user, platform operator, platform implementor, and/or project contributor.
+- Explaining the feature largely in terms of examples.
+- If applicable, provide sample error messages, deprecation warnings, or migration guidance.
+- If applicable, describe the differences between teaching this to existing users and new users.
+
+# How it Works
+[how-it-works]: #how-it-works
+
+Ideally, the user should run something like `pack detect --path ./path/to/project --builder builder:name` and it should run only the detect binary in the lifecycle and output the logs / output of it. Preferrably, can also output the `group.toml` to a directory specified with `--detect-output-dir`.
+
+The following flags should be supported and they will work more or less like `pack build`.
+
+|Short|Long|type|
```

Comment:
Probably also needed:

```
  -d, --descriptor string            Path to the project descriptor file
      --docker-host string           Address to docker daemon that will be exposed to the build container.
                                     If not set (or set to empty string) the standard socket location will be used.
                                     Special value 'inherit' may be used in which case DOCKER_HOST environment variable will be used.
                                     This option may set DOCKER_HOST environment variable for the build container if needed.
      --gid int                      Override GID of user's group in the stack's build and run images. The provided value must be a positive number
      --lifecycle-image string       Custom lifecycle image to use for analysis, restore, and export when builder is untrusted.
      --network string               Connect detect and build containers to network
      --pull-policy string           Pull policy to use. Accepted values are always, never, and if-not-present. (default "always")
      --run-image string             Run image (defaults to default stack's run image)
      --uid int                      Override UID of user in the stack's build and run images. The provided value must be a positive number
      --workspace string             Location at which to mount the app dir in the build image
```

* `descriptor` controls things like environment and buildpacks to use
* `docker-host` for accessing the run image
* `lifecycle-image` we need to run the `analyze` step
* `gid` and `uid` could maybe affect how buildpack `./bin/detect` behaves
* `network` and `workspace` for configuring the container environment where buildpacks' `./bin/detect` are run
* `pull-policy` and `run-image` for influencing the run image that actually gets used